compiler/test/compilable/test23145.d

```diff
@@ -0,0 +1,37 @@
+/* TEST_OUTPUT:
+---
+compilable/test23145.d(117): Deprecation: `new` constructor must be annotated with `scope` to initializer `scope` variable `c`
```

Comment:
```suggestion
compilable/test23145.d(117): Deprecation: `scope` allocation requires that `$FULLY_QUALIFIED_NAME` constructor be annotated with `scope`
```

And an `errorSupplemental` saying "Add `scope` here" with the file/line of the ctor

Reply:
cc @WalterBright